### PR TITLE
[OBS] API Updated

### LINKED ISSUE #431: [OBS] API Updated
https://confluence.tsi-dev.otc-service.com/display/DR/RM-4339-OBS-API

### PR DESCRIPTION
**Pull request by Aloento**
Updated the APIs and organized the file structure.
No other changes.

fix #431 

```
=== RUN   TestObsBucketLifecycle
--- PASS: TestObsBucketLifecycle (1.99s)
=== RUN   TestObsObjectLifecycle
--- PASS: TestObsObjectLifecycle (1.83s)
PASS
```

### CODE REVIEW

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

File: openstack/obs/client_base.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		info[0] = fmt.Sprintf("[OBS SDK Version=%s", obs_sdk_version)
+		info[1] = fmt.Sprintf("Endpoint=%s", conf.endpoint)
+		accessMode := "Virtual Hosting"
+		if conf.pathStyle {
+			accessMode = "Path"
+		}
+		info[2] = fmt.Sprintf("Access Mode=%s]", accessMode)
+		doLog(LEVEL_WARN, strings.Join(info, "];["))
+	}
+	doLog(LEVEL_DEBUG, "Create obsclient with config:\n%s\n", conf)
```
Comment (anton-sidelnikov) [inline openstack/obs/client_base.go:43]: doLog using only for obs, can you change it to some standard method?

File: openstack/obs/transfer.go [hunk 2]
```diff
@@ -0,0 +1,830 @@
...
+
+func getCheckpointFile(ufc *UploadCheckpoint, uploadFileStat os.FileInfo, input *UploadFileInput, obsClient *ObsClient) (needCheckpoint bool, err error) {
+	checkpointFilePath := input.CheckpointFile
+	checkpointFileStat, err := os.Stat(checkpointFilePath)
+	if err != nil {
+		doLog(LEVEL_DEBUG, fmt.Sprintf("Stat checkpoint file failed with error: [%v].", err))
+		return true, nil
+	}
+	if checkpointFileStat.IsDir() {
+		doLog(LEVEL_ERROR, "Checkpoint file can not be a folder.")
```
Comment (anton-sidelnikov) [inline openstack/obs/transfer.go:137]: again doLog

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]
Also please look at latest documentation and add missing methods. It will be perfect if you add more tests for obs

File: openstack/obs/client_base.go (hunk 1 above)
Comment (Aloento) [reply openstack/obs/client_base.go:43]: https://github.com/opentelekomcloud/gophertelekomcloud/blob/665dacb4afbcc8fde38ae6edb50e4558014e7f8f/openstack/obs/log.go#L266

doLog contains Log formatting methods customized for OBS.
Do you have any better ideas?

**Review by anton-sidelnikov — APPROVED** [review]

**Review by artem-lifshits — APPROVED** [review]

**Pull request merged**